File: src/main/java/meowing/zen/feats/meowing/meowsounds.java
package meowing.zen.feats.meowing;

import meowing.zen.utils.EventBus;
import meowing.zen.utils.EventTypes;
import meowing.zen.featManager;
import meowing.zen.utils.utils;
import net.minecraft.sound.SoundEvents;

public class meowsounds {
    public static void initialize() {
        featManager.register(new meowsounds(), () -> {
            EventBus.register(EventTypes.GameMessageEvent.class, meowsounds.class, meowsounds::handleGameMessage);
        });
    }

    private static void handleGameMessage(EventTypes.GameMessageEvent event) {
        String content = event.message.getString().toLowerCase();
        if (content.contains("meow")) utils.playSound(SoundEvents.ENTITY_CAT_AMBIENT, 0.8f, 1.0f);
    }
}

File: src/main/kotlin/meowing/zen/feats/meowing/meowsounds.kt
package meowing.zen.feats.meowing

import meowing.zen.feats.Feature
import meowing.zen.Zen.Companion.mc
import meowing.zen.events.ChatEvent
import meowing.zen.utils.Utils.removeFormatting
import net.minecraft.sound.SoundEvents

object meowsounds : Feature("meowsounds") {
    private val meowRegex = Regex("(?:Guild|Party|Co-op|From|To)? ?>? ?(?:\\[.+?])? ?[a-zA-Z0-9_]+ ?(?:\\[.+?])?: (.+)")

    override fun initialize() {
        register<ChatEvent.Receive> {
            val content = it.message.string.removeFormatting().lowercase()
            val match = meowRegex.find(content) ?: return@register
            if (match.groups[1]?.value?.contains("meow", ignoreCase = true) != true) return@register

            mc.player?.let { player ->
                mc.world?.playSound(
                    null,
                    player.pos.x, player.pos.y, player.pos.z,
                    SoundEvents.ENTITY_CAT_AMBIENT,
                    net.minecraft.sound.SoundCategory.AMBIENT,
                    0.8f, 1.0f
                )
            }
        }
    }
}